### PR TITLE
Custom auto-setup temporal docker image

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Make the update of temporal possible.

## How
This is a custom auto-setup script for the temporal environment. Unfortunately there is no other way properly update the DB without copy pasting parts of the temporal auto-setup script. Ideally temporal would provide a dedicated container for it DB but it is not the case right now.

### CODE REVIEW

**Comment by benmoriceau** [comment]
Moving it to draft, because I need to merge the removal of the docker-compose.

**Review by cgardens — COMMENTED** [review]
cool! we had talked about just dropping the db for now. i take it from this PR you think we should just go straight for the upgrade solution instead?

**Comment by benmoriceau** [comment]
> t just dropping the db for now. i take it from this PR you think we should just go straight for the upgrade solution instead?

I don't think that dropping the DB will be needed here the upgrade will create the DB if needed. However the PR is affected by the revert of the removal of the docker-compose.

**Comment by benmoriceau** [comment]
Manual test run:

**Review by jrhizor — CHANGES_REQUESTED** [review]
Added some comments.

My main worry is around upgrading to a new Temporal version after this. Do we need to update the update script itself to be able to do that? Is there a way to detect incompatibilities via automated testing?

**Comment by benmoriceau** [comment]
I didn't want t o push. I put it back in draft.

**Comment by cgardens** [comment]
Nice. Agreed with @jrhizor 's comments. As long as this is something we think is going to reasonable to maintain in future temporal versions, this seeems like a good approach.

File: airbyte-temporal/build.gradle [hunk 1]
```diff
@@ -0,0 +1,10 @@
+task copyScripts(type: Copy) {
```
[the remark this answers is not in the document]
Comment (benmoriceau) [reply airbyte-temporal/build.gradle:1]: The docker build task will copy the dockerfile to the `build/docker` folder so I need to copy the script in this folder too.

File: airbyte-temporal/scripts/update-and-start-temporal.sh [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+SCHEMA_DIR=${TEMPORAL_HOME}/schema/postgresql/v96/temporal/versioned
+VISIBILITY_SCHEMA_DIR=${TEMPORAL_HOME}/schema/postgresql/v96/visibility/versioned
+
+SKIP_DEFAULT_NAMESPACE_CREATION="${SKIP_DEFAULT_NAMESPACE_CREATION:-false}"
+DEFAULT_NAMESPACE="${DEFAULT_NAMESPACE:-default}"
+DEFAULT_NAMESPACE_RETENTION=${DEFAULT_NAMESPACE_RETENTION:-1}
+
+# See https://github.com/temporalio/temporal/blob/release/v1.13.x/docker/entrypoint.sh
+initEntryPoint() {
```
[the remark this answers is not in the document]
Comment (benmoriceau) [reply airbyte-temporal/scripts/update-and-start-temporal.sh:21]: Done

File: docker-compose.yaml [hunk 3]
```diff
@@ -176,12 +176,11 @@ services:
       - OPENREPLAY=${OPENREPLAY:-}
       - PAPERCUPS_STORYTIME=${PAPERCUPS_STORYTIME:-}
   airbyte-temporal:
-    image: temporalio/auto-setup:1.7.0
+    image: airbyte/temporal:${VERSION}
     logging: *default-logging
     container_name: airbyte-temporal
     restart: unless-stopped
-    ports:
-      - 7233:7233
```
[the remark this answers is not in the document]
Comment (benmoriceau) [reply docker-compose.yaml:184]: Done

File: docker-compose.yaml [hunk 4]
```diff
@@ -176,12 +176,11 @@ services:
...
+    command: ./update-and-start-temporal.sh
```
[the remark this answers is not in the document]
Comment (benmoriceau) [reply docker-compose.yaml:183]: Thanks, I have set it as an entry point.

**Comment by benmoriceau** [comment]
> Added some comments.
> 
> My main worry is around upgrading to a new Temporal version after this. Do we need to update the update script itself to be able to do that? Is there a way to detect incompatibilities via automated testing?

I added a test, It is hacky and based on the fact that the docker compose up takes less than 1 minute.

**Review by jrhizor — APPROVED** [review]

**Pull request merged**